BigQuery/src/Dataset.php

```diff
@@ -379,4 +379,75 @@ public function identity()
     {
         return $this->identity;
     }
+
+    /**
+     * Lazily instantiates a machine learning model in the dataset. There are no
+     * network requests made at this point. To see the operations that can be performed on a
+     * model, please see {@see Google\Cloud\BigQuery\Model}.
+     *
+     * Example:
+     * ```
+     * $model = $dataset->model('my_model');
+     * echo $model->id();
+     * ```
+     *
+     * @param $id string The model's ID.
+     * @return Model
+     */
+    public function model($id)
+    {
+        return new Model(
+            $this->connection,
+            $id,
+            $this->id(),
+            $this->identity['projectId']
+        );
+    }
+
+    /**
+     * Fetches all of the models in the dataset.
+     *
+     * Example:
+     * ```
+     * $models = $dataset->models();
+     *
+     * foreach ($models as $model) {
+     *     echo $model->id() . PHP_EOL;
+     * }
+     * ```
+     *
+     * @see https://cloud.google.com/bigquery/docs/reference/v2/models/list Models list API documentation.
+     *
+     * @param array $options [optional] {
+     *     Configuration options.
+     *
+     *     @type int $maxResults Maximum number of results to return per page.
+     *     @type int $resultLimit Limit the number of results returned in total.
+     *           **Defaults to** `0` (return all results).
+     *     @type string $pageToken A previously-returned page token used to
+     *           resume the loading of results from a specific point.
+     * }
+     *
+     * @return ItemIterator<Model>
+     */
+    public function models(array $options = [])
+    {
+        $resultLimit = $this->pluck('resultLimit', $options, false);
+        $datasetId = $this->id();
+        $projectId = $this->identity['projectId'];
+
+        return new ItemIterator(
+            new PageIterator(
+                function (array $model) {
+                    return $this->model($model['modelReference']['modelId']);
```

Comment:
Hey John, I ran into this one while developing. The data that is returned form the List call is not the full set of data that is returned when fetching the individual model resource. It seemed better to not include it so that you would consistently have the same data. In that case, should I just ditch it from the constructor and allow it to hydrate lazily?

Reply:
I removed adding info from the constructor for now since it should always be triggered via the reload method. I'll keep an eye on this if the BQ team modifies the Models return response to include the full set of data.

Reply:
Makes sense. Yeah, I think we can probably remove the argument if it’s unused. It’s nothing to add it back later if the situation changes.

Reply:
The [models.list documentation](https://cloud.google.com/bigquery/docs/reference/rest/v2/models/list) says the following:

```
Only the following fields are populated: modelReference, modelType, creationTime, lastModifiedTime and labels.
```

Which leads me to believe this will be a permanent decision to only return a subset of the information on list calls. Since the information returned appears useful to users, I'd like to propose we populate the `Model` object with the information when we have it, with documentation to let users know only a subset of the data is returned and to call `reload` on the `Model` if they would like access to all of the information.